src/lib/tls/tls12/tls_client_impl_12.cpp

```diff
@@ -61,19 +61,20 @@ class Client_Handshake_State_12 final : public Handshake_State
 /*
 * TLS 1.2 Client  Constructor
 */
-Client_Impl_12::Client_Impl_12(Callbacks& callbacks,
-                               Session_Manager& session_manager,
-                               Credentials_Manager& creds,
-                               const Policy& policy,
-                               RandomNumberGenerator& rng,
-                               const Server_Information& info,
+Client_Impl_12::Client_Impl_12(std::shared_ptr<Callbacks> callbacks,
+                               std::shared_ptr<Session_Manager> session_manager,
+                               std::shared_ptr<Credentials_Manager> creds,
+                               std::shared_ptr<const Policy> policy,
+                               std::shared_ptr<RandomNumberGenerator> rng,
+                               Server_Information info,
                                bool datagram,
-                               const std::vector<std::string>& next_protocols,
+                               std::vector<std::string> next_protocols,
```

Comment:
a) I have a feeling clang-tidy will complain about this
b) I don't really understand the motivation for the change here